File: Core/YABACore/src/commonMain/kotlin/dev/subfly/yabacore/state/home/HomeUIState.kt
package dev.subfly.yabacore.state.home

import dev.subfly.yabacore.model.ui.BookmarkUiModel
import dev.subfly.yabacore.model.ui.HomeFolderRowUiModel
import dev.subfly.yabacore.model.ui.TagUiModel
import dev.subfly.yabacore.model.utils.BookmarkAppearance
import dev.subfly.yabacore.model.utils.CardImageSizing
import dev.subfly.yabacore.model.utils.SortOrderType
import dev.subfly.yabacore.model.utils.SortType

data class HomeUIState(
    /** Visible folder rows for Home, already flattened and ordered. */
    val folderRows: List<HomeFolderRowUiModel> = emptyList(),
    /** Expanded folder IDs (Home-only; not persisted). */
    val expandedFolderIds: Set<String> = emptySet(),
    val tags: List<TagUiModel> = emptyList(),
    val recentBookmarks: List<BookmarkUiModel> = emptyList(),
    val bookmarkAppearance: BookmarkAppearance = BookmarkAppearance.LIST,
    val cardImageSizing: CardImageSizing = CardImageSizing.SMALL,
    val collectionSorting: SortType = SortType.CREATED_AT,
    val sortOrder: SortOrderType = SortOrderType.ASCENDING,
    val isLoading: Boolean = true,
)

File: Darwin/YABACore/State/Home/HomeUIState.swift
//
//  HomeUIState.swift
//  YABACore
//

import Foundation

public struct HomeUIState: Sendable {
    public var isLoading: Bool
    public var bookmarkAppearance: BookmarkAppearance
    public var cardImageSizing: CardImageSizing
    public var collectionSortType: SortType
    public var sortOrder: SortOrderType
    /// Folder rows expanded in the tree UI.
    public var expandedFolderIds: Set<String>

    public init(
        isLoading: Bool = true,
        bookmarkAppearance: BookmarkAppearance = .list,
        cardImageSizing: CardImageSizing = .small,
        collectionSortType: SortType = .editedAt,
        sortOrder: SortOrderType = .descending,
        expandedFolderIds: Set<String> = []
    ) {
        self.isLoading = isLoading
        self.bookmarkAppearance = bookmarkAppearance
        self.cardImageSizing = cardImageSizing
        self.collectionSortType = collectionSortType
        self.sortOrder = sortOrder
        self.expandedFolderIds = expandedFolderIds
    }
}
